### PR TITLE
SDK-329 [Java SDK] update README with crypto refactor changes

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-329](https://incountry.atlassian.net/browse/SDK-329)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -172,76 +177,67 @@ public interface SecretKeyAccessor {
...
 
 You can implement `SecretKeyAccessor` interface and pass secrets/keys in multiple ways:
 
 1. As a constant SecretsData object
-    ```java
-    SecretsData secretsData = new SecretsData(secretsList, currentVersion);
-    SecretKeyAccessor accessor = () -> secretsData;
-    ```
+```java
+SecretsData secretsData = new SecretsData(secretsList, currentSecret);
```
Comment (RommeroInCountry) [inline README.md:204]: don't forget to update AWS example - it has old constructor for SecretsData

File: README.md [hunk 2]
```diff
@@ -897,88 +921,65 @@ public void test() {
...
+provider.registerCipher(anotherCipher);
+StorageConfig config = new StorageConfig();
+//...
+config.setCryptoProvider(provider);
+Storage storage = StorageImpl.newStorage(config);
+
 ```
 
-For using of custom encryption you need to implement the following interface:
+For using of custom encryption you need to create an implementation the following abstract class:
```
Comment (RommeroInCountry) [inline README.md:938]: To use custom encryption you need to implement the following abstract class:

File: README.md [hunk 3]
```diff
@@ -997,65 +998,15 @@ secrets_data = {
 ```
 ---
 
-`version` attribute is used to differ one custom encryption from another and from the default encryption as well.
-This way SDK will be able to successfully decrypt any old data if encryption changes with time.
+Parameter `name` of `AbstractCipher` constructor is used to differ one custom encryption from another and from the default encryption as well. This way SDK will be able to successfully decrypt any old data if encryption changes with time.
 
-`isCurrent` attribute allows to specify one of the custom encryption implementations that will be used for encryption. Only one implementation can be set as `isCurrent() == true`.
+You can set current cipher implementation via a constructor of `CryptoProvider`. Such cipher will be used for encryption.
```
Comment (RommeroInCountry) [inline README.md:1003]: Such cipher -> This cipher will then be used for encryption as well

File: README.md [hunk 4]
```diff
@@ -997,65 +998,15 @@ secrets_data = {
...
 
-If none of the configurations have `isCurrent() == true` then the SDK will use default encryption to encrypt stored data. At the same time it will keep the ability to decrypt old data, encrypted with custom encryption (if any).
+If none of the custom implementations of `AbstractCipher` then the SDK will use default encryption to encrypt stored data. At the same time it will keep the ability to decrypt old data, encrypted with custom encryption (if any).
```
Comment (RommeroInCountry) [inline README.md:1005]: If you don't override default cipher (by providing a new one to `CryptoProvider` constructor) the the SDK ...

File: README.md (hunk 3 above)
Comment (RommeroInCountry) [reply README.md:1003]: You can override default cipher by providing a new one to CryptoProvider constructor. This cipher will then be used for encryption as well.

File: README.md [hunk 5]
```diff
@@ -997,65 +998,15 @@ secrets_data = {
...
 
-Here's an example of how you can set up SDK to use custom encryption (using Fernet encryption from https://github.com/l0s/fernet-java8 )
+You can see an example of an implementation of custom encryption (using Fernet encryption from https://github.com/l0s/fernet-java8) by [following this link](/src/test/java/com/incountry/residence/sdk/crypto/testimpl/FernetCipher.java)
```
Comment (RommeroInCountry) [inline README.md:1007]: You can see an example of custom encryption implementation

**Review by RommeroInCountry — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (VladimirLevitskiy) [reply README.md:204]: AWS example is the source file, the compiler doesn't allow to leave the invalid source code

**Review by robert-par — APPROVED** [review]

**Review by RommeroInCountry — APPROVED** [review]

**Pull request merged**